### ISSUE TITLE
✅ 📚 🐞 Bug Report: Fallback for related posts not working in the Posts widget - [ED-10701]

### ISSUE THREAD

**Issue by trinermedia** [elementor/elementor#21436]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Posts are not displayed in the [Posts widget](https://elementor.com/help/posts-widget-pro/) when the query is set to "Related" and there are no posts of the same term, even if "Recent Posts" is selected as fallback.

### Steps to reproduce

1. Create some posts. Make sure, one of them has only one category assigned and its the only post of this category
![posts php](https://user-images.githubusercontent.com/86961599/220846776-6e9f89ad-2d02-4ef5-a0ba-52eaac4bfad5.png)
2. Create a single post template and add the posts widget. Set the query to "Related", choose "Include by Term" > "Cateogries" and set the fallback option to "Recent posts"
![single post template](https://user-images.githubusercontent.com/86961599/220846802-8e04c4b6-596a-4054-89eb-fa0dbfd25e8a.png)
3. View the post which is the only one of his category and look for related posts.
![view post a1](https://user-images.githubusercontent.com/86961599/220846814-6cbccb64-b161-4dfa-a256-3e329de9157c.png)
![view post b1](https://user-images.githubusercontent.com/86961599/220846821-39759eb2-1afb-4698-85b1-82e804cbd14e.png)

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt

					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v10.3.37
	PHP Version: 8.0.27
	PHP Memory Limit: 256M
	PHP Max Input Vars: 200000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: https://sandbox.triner.ch/debugging
	Home URL: https://sandbox.triner.ch/debugging
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Admin Email: webteam@triner.ch
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.6.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.11.2
		Author: Elementor.com

	Elementor Pro
		Version: 3.11.2
		Author: Elementor.com


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive by default
	Upgrade Swiper Library: Active by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor V2: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Save as Default: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Mega Menu: Inactive by default


== Log ==
	
Log: showing 5 of 52023-02-23 07:26:04 [info] elementor-pro::elementor_pro_updater Started 
2023-02-23 07:26:04 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-02-23 07:26:04 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-02-23 07:26:04 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.1',
  'to' => '3.11.2',
)]
2023-02-23 07:26:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.1',
  'to' => '3.11.2',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	

				
````
</details>

**Comment by nicholaszein** [comment]
Hello @trinermedia! 

Thank you for submitting your issue! 🙏 

Please allow us some time to investigate it, and get back to you with our findings. 🧐

Best regards

**Comment by jfama** [comment]
Bumped, encountering the same issue on our site.

**Comment by DianeRt** [comment]
Bumped, encountering the same issue on our site.

**Comment by berrytimmermans** [comment]
Bumped, encountering the same issue on our site.

**Comment by DiiiaZoTe** [comment]
Bumped, encountering the same issue on our site.

**Comment by hyperjit** [comment]
Bump! Same issue. Any solution yet?

**Comment by cartesim01** [comment]
Same here, solution?

**Comment by jfama-purecobalt** [comment]
I think I may have found a partial workaround.

In elementor-pro/modules/query-control/classes/elementor-related-query.php, line 60:
`$this->fallback_args['no_found_rows'] = true;`

I don't fully understand what value this adds to the fallback query. Disabling it or setting it to false resolved the issue for me and fallback queries work as expected.

**Comment by jfama-purecobalt** [comment]
Looking at it, the get_fallback_query function should be updated with some adjustments:
Lines 57-61:
```
		$this->fallback_args['no_found_rows'] = true;

		$this->fallback_args = apply_filters( 'elementor/query/fallback_query_args', $this->fallback_args, $this->widget, $original_query->query_vars );

		return new \WP_Query( $this->fallback_args );
```		

This allows two improvements:
- Being able to set `no_found_rows` as needed for my query, in the filter.
- Exposing the original query arguments to the filter. This could be useful in a number of cases. As an example, if my original query excluded a specific post dynamically, and my fallback query shows latest posts, I still may want to exclude that post from the original query.

**Comment by nicholaszein** [comment]
Hello everyone! 

We are aware of this issue and our team is already working on a fix. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉

Best regards

**Comment by nicholaszein** [comment]
Hey everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.16! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
